### PR TITLE
Add new langauge evolution page

### PR DESCRIPTION
**Pull request by mit-mit**

### CODE REVIEW

**Comment by kwalrath** [comment]
I'm reviewing the contents, but I'd love to hear more about what drove the desire for this page, and how we expect to use it.

**Comment by mit-mit** [comment]
>  I'd love to hear more about what drove the desire for this page

My goal was to provide an overview of how we're evolving, with the hope of demonstrating that Dart is a vibrant language.

This is similar to:
https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-version-history

**Comment by mit-mit** [comment]
Edits and code change LGTM

**Comment by kevmoo** [comment]
Cool! https://dart.dev/guides/language/evolution

File: src/_guides/language/evolution.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+Dart has always supported literal lists and maps, but
+Dart 2.2 added support for **[set literals][]**:
+
+```dart
+const Set<String> currencies = {'EUR', 'USD', 'JPY'};
+```
+## Dart 2.3
+
+Dart 2.3 added three operators designed to improve code that performs
+list manipulation, such as declarative UI code.
```
Comment (leafpetersen) [inline src/_guides/language/evolution.md]: Not just lists - all primitive collections (lists, sets, maps).

Comment (kwalrath) [reply src/_guides/language/evolution.md]: How about I change it to this:

...improve code (such as declarative UI code) that manipulates lists, sets, and maps.

Comment (leafpetersen) [reply src/_guides/language/evolution.md]: Works for me.

Comment (kwalrath) [reply src/_guides/language/evolution.md]: Or maybe "lists, sets, and maps" -> "collections" or "primitive collections"

**Pull request merged**